File: backend/src/main/java/com/project2/domain/member/repository/FollowRepository.java
package com.project2.domain.member.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.project2.domain.member.entity.Follows;
import com.project2.domain.member.entity.Member;

@Repository
public interface FollowRepository extends JpaRepository<Follows, Long> {
	Optional<Follows> findByFollowerAndFollowing(Member follower, Member following);

	Page<Follows> findByFollowing(Member following, Pageable pageable);

	List<Follows> findByFollowing(Member following);

	List<Follows> findByFollower(Member follower);

	long countByFollower(Member follower);

	long countByFollowing(Member member);

	boolean existsByFollowerAndFollowing(Member follower, Member following);

	void deleteByFollowerAndFollowing(Member follower, Member following);
}

File: backend/src/main/java/com/project2/domain/member/repository/FollowRepository.kt
package com.project2.domain.member.repository

import com.project2.domain.member.entity.Follows
import com.project2.domain.member.entity.Member
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface FollowRepository : JpaRepository<Follows?, Long?> {
    fun findByFollowerAndFollowing(follower: Member?, following: Member?): Optional<Follows?>?

    fun findByFollowing(following: Member?, pageable: Pageable?): Page<Follows?>?

    fun findByFollowing(following: Member?): List<Follows?>?

    fun findByFollower(follower: Member?): List<Follows?>?

    fun countByFollower(follower: Member?): Long

    fun countByFollowing(member: Member?): Long

    fun existsByFollowerAndFollowing(follower: Member?, following: Member?): Boolean

    fun deleteByFollowerAndFollowing(follower: Member?, following: Member?)
}